File: lab/react/vite-lab1/src/app/ibuki.ts
import { filter, debounceTime } from 'rxjs/operators'
import { Subject, BehaviorSubject } from 'rxjs'

const subject = new Subject()
const behSubject = new BehaviorSubject<any>(0)

export function ibukiEmit(id: string, options: any) {
    subject.next({ id: id, data: options })
}

export function ibukiFilterOn(id: string) {
    return subject.pipe(filter((d: any) => d.id === id))
}

export function ibukiHotEmit(id: string, options: any) {
    behSubject.next({ id: id, data: options })
}

export function ibukiHotFilterOn(id: string) {
    return behSubject.pipe(filter((d: any) => d.id === id))
}

export function ibukiDdebounceEmit(id: string, options: any) {
    subject.next({ id: id, data: options })
}

export function ibukiDebounceFilterOn(id: string, debouncePeriod: number = 1000) {
    return subject
        .pipe(filter((d: any) => d.id === id))
        .pipe(debounceTime(debouncePeriod))
}


// export { emit, filterOn, hotEmit, hotFilterOn, debounceEmit, debounceFilterOn }

File: lab/Flutter/trace_mobile/lib/common/classes/ibuki.dart
import 'package:rxdart/rxdart.dart';

class Ibuki {
  // static BehaviorSubject<Map<String, dynamic>> subject = BehaviorSubject<Map<String, dynamic>>();
  static PublishSubject<Map<String, dynamic>> subject = PublishSubject<Map<String, dynamic>>();
  static void emit(String id, dynamic data){
    subject.add({"id":id, "data": data});
  }

  static Stream<Map<String, dynamic>> filterOn(String id){
    return(subject.stream.where((d) => d['id'] == id));
  }

  static void debounceEmit(String id, dynamic data){
    subject.add({"id":id, "data": data});
  }

  static Stream<Map<String, dynamic>> debounceFilterOn(String id, {int debouncePeriod = 1000}){
    return(subject.stream.where((d) => d['id'] == id)).debounceTime(Duration(milliseconds: debouncePeriod));
  }
}

class IbukiKeys{
  static String productFilerKey = 'IBUKI-PRODUCT-FILTER';
}